docs/book/src/developer/architecture/controllers/machine-pool.md

```diff
@@ -0,0 +1,114 @@
+# Machine  Controller
+
+![](../../../images/cluster-admission-machinepool-controller.png)
+
+The Machine controller's main responsibilities are:
+
+* Setting an OwnerReference on:
+    * Each MachinePool object to the Cluster object.
+    * The associated BootstrapConfig object.
+    * The associated InfrastructureMachinePool object.
+* Copy data from `BootstrapConfig.Status.DataSecretName` to `MachinePool.Spec.Template.Spec.Bootstrap.DataSecretName` if
+`MachinePool.Spec.Template.Spec.Bootstrap.DataSecretName` is empty.
+* Setting NodeRefs to be able to associate instances and kubernetes nodes.
+* Deleting Nodes in the target cluster when the associated instance is deleted.
+* Cleanup of related objects.
```

Comment:
What are the related objects? Is it referring to things like BootstrapConfig?

Reply:
removed, this doesn't apply here. There is one bootstrap config per machine pool (not one per instance)